here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/ops/PropertyUtil.java

```diff
@@ -0,0 +1,220 @@
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.payload.events.QueryDelimiter;
+import com.here.naksha.lib.core.models.payload.events.QueryOperation;
+import com.here.naksha.lib.core.models.payload.events.QueryParameter;
+import com.here.naksha.lib.core.models.payload.events.QueryParameterList;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.ValueList;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class PropertyUtil {
+
+  private static final String NULL_PROP_VALUE = ".null";
+
+  /**
+   * Function builds Property Operation (POp) based on property key:value pairs supplied as API query parameter.
+   * We iterate through all the parameters, exclude the keys that match with provided excludeKeys,
+   * and interpret the others by identifying the desired operation.
+   * <p>
+   * Multiple parameter keys result into AND list.
+   * <br>
+   * So, "p.prop_1=value_1&p.prop_2=value_2" will form AND condition as (p.prop_1=value_1 AND p.prop_2=value_2).
+   * </p>
+   *
+   * <p>
+   * Multiple parameter values concatenated with "," (COMMA) delimiter, will result into OR list.
+   * <br>
+   * So, "p.prop_1=value_1,value_11" will form OR condition as (p.prop_1=value_1 OR p.prop_1=value_11).
+   * </p>
+   *
+   * @param queryParams API query parameter from where property search params need to be extracted
+   * @param excludeKeys List of param keys to be excluded (i.e. not part of property search)
+   * @return POp property operation that can be used as part of {@link ReadRequest}
+   */
+  public static @Nullable POp buildOperationForPropertySearchParams(
+      final @Nullable QueryParameterList queryParams, final @Nullable List<String> excludeKeys) {
+    if (queryParams == null) return null;
+    // global initialization
+    final List<POp> globalOpList = new ArrayList<>();
+    // iterate through each parameter
+    for (final QueryParameter param : queryParams) {
+      // extract param key, operation, values, delimiters
+      final String key = param.key();
+      final QueryOperation operation = param.op();
+      final ValueList values = param.values();
+      final List<QueryDelimiter> delimiters = param.valuesDelimiter();
+
+      // is this key to be excluded?
+      if (excludeKeys != null && excludeKeys.contains(key)) continue;
+      // prepare property search operation
+      final POp crtOp = preparePropertySearchOperation(operation, key, values, delimiters);
+      // add current search operation to global list
+      globalOpList.add(crtOp);
+    }
+
+    if (globalOpList.isEmpty()) return null;
+    // return single operation or AND list (in case of multiple operations)
+    final POp[] allPOpArr = globalOpList.toArray(POp[]::new);
+    return (allPOpArr.length > 1) ? POp.and(allPOpArr) : allPOpArr[0];
```

Comment:
nth / picky (😄 ): we can check the size of `globalOpList` earlier and skip unnecessary array allocation - so that `globalopList.toArray` will happen only when we're sure that we are dealing with more than 1 item